packages/client/src/pages/Forum/elements/Topic/Topic.tsx

```diff
@@ -1,23 +1,67 @@
-import { FC } from 'react'
+import { FC, useCallback, useState } from 'react'
 import { Link } from 'react-router-dom'
 import styles from './topic.module.css'
 import { Topics } from '@store/forum/types'
+import { SmileIcon } from '@assets/icons'
+import EmojiList from '@pages/ForumTopic/elements/EmojiList'
+import { useAppSelector } from '@store/hooks'
+import { forumSelectors } from '@store/forum'
 
 const Topic: FC<Topics> = ({
     id,
     title,
     description,
     createdBy,
     createdAt,
-}) => (
-    <Link to={id.toString()} className={styles.root}>
-        <div className={styles.title}>{title}</div>
-        {description && <div className={styles.description}>{description}</div>}
-        <div className={styles.meta}>
-            <span className={styles.author}>{createdBy}</span>
-            <span className={styles.date}>{createdAt}</span>
-        </div>
-    </Link>
-)
+    emoji,
+}) => {
+    const [showEmojiList, setShowEmojiList] = useState(false)
+    const topicEmoji = useAppSelector(forumSelectors.selectixEmojis)
+    const handleEmojiClick = useCallback(
+        (event: React.MouseEvent<HTMLElement>) => {
+            event.preventDefault()
+            setShowEmojiList(prevState => !prevState)
+        },
+        []
+    )
+
+    const handleMouseLeave = useCallback(() => {
+        setShowEmojiList(false)
+    }, [])
+    return (
+        <Link
+            to={id.toString()}
+            className={styles.root}
+            onMouseLeave={handleMouseLeave}
+        >
+            <div className={styles.header}>
+                <div className={styles.title}>{title}</div>
+                <div className={styles.emojiContainer}>
+                    <SmileIcon
+                        className={styles.emojiButton}
+                        onClick={handleEmojiClick}
+                    />
+                    {showEmojiList && <EmojiList topicId={id} />}
+                </div>
+            </div>
+            {description && (
+                <div className={styles.description}>{description}</div>
+            )}
+            <div className={styles.meta}>
+                <span className={styles.author}>{createdBy}</span>
+                <span className={styles.date}>{createdAt}</span>
+            </div>
+            {emoji?.length && (
```

Comment:
Числовые значения в таком виде лучше явно приводить к boolean, иначе при нулевой длине выведется `0`. Можно сравнением `emoji?.length > 0` или проще всего так:

```suggestion
            {Boolean(emoji?.length) && (
```